src/_utils.js

```diff
@@ -28,22 +28,41 @@ export const addClassName = (path, jsxId) => {
   // Find className and collect spreads
   for (let i = attributes.length - 1, attr; (attr = attributes[i]); i--) {
     const node = attr.node
+
     if (t.isJSXSpreadAttribute(attr)) {
-      const name = node.argument.name
-      const attrNameDotClassName = t.memberExpression(
-        t.isMemberExpression(node.argument)
-          ? node.argument
-          : t.identifier(name),
-        t.identifier('className')
-      )
+      if (t.isObjectExpression(node.argument)) {
+        const properties = node.argument.properties
 
-      spreads.push(
-        // `${name}.className != null && ${name}.className`
-        and(
-          t.binaryExpression('!=', attrNameDotClassName, t.nullLiteral()),
-          attrNameDotClassName
+        const index = properties.findIndex(
+          property => property.key.name === 'className'
         )
-      )
+
+        if (~index) {
+          className = attr.get('argument').get(`properties.${index}`)
+          break
+        }
+      }
+
+      if (
+        t.isMemberExpression(node.argument) ||
+        t.isIdentifier(node.argument)
```

Comment:
is this if necessary? what else can we have here? maybe you can add a `continue` inside of `if (t.isObjectExpression(node.argument)) {`

Reply:
> what else can we have here?

You mean other conditions?

Reply:
yea I mean we can spread:

* ObjectExpression
* MemberExpression
* Identifier

so I was thinking that you can add a `continue` inside of the `isObjectExpression` if statement body, this way you don't need this other if statement.

Probably there are other weird edge cases like an inline function call like `{...foo()}` but shrug :P otherwise your code is fine if we want to keep the safety check :) Your call I am fine either ways ;)

Reply:
Understand. If you want I can add function call support too?